data/deduplicator/deduplicator/data_set_delete_origin_older.go

```diff
@@ -0,0 +1,83 @@
+package deduplicator
+
+import (
+	"context"
+	"slices"
+
+	"github.com/tidepool-org/platform/data"
+	dataStore "github.com/tidepool-org/platform/data/store"
+	dataTypesBolus "github.com/tidepool-org/platform/data/types/bolus"
+	dataTypesFood "github.com/tidepool-org/platform/data/types/food"
+	dataTypesUpload "github.com/tidepool-org/platform/data/types/upload"
+	"github.com/tidepool-org/platform/pointer"
+)
+
+const (
+	DataSetDeleteOriginOlderName    = "org.tidepool.deduplicator.dataset.delete.origin.older"
+	DataSetDeleteOriginOlderVersion = "1.0.0"
+)
+
+type DataSetDeleteOriginOlder struct {
+	*DataSetDeleteOriginBase
+}
+
+func NewDataSetDeleteOriginOlder() (*DataSetDeleteOriginOlder, error) {
+	dataSetDeleteOriginBase, err := NewDataSetDeleteOriginBase(DataSetDeleteOriginOlderName, DataSetDeleteOriginOlderVersion, &dataSetDeleteOriginOlderProvider{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &DataSetDeleteOriginOlder{
+		DataSetDeleteOriginBase: dataSetDeleteOriginBase,
+	}, nil
+}
+
+type dataSetDeleteOriginOlderProvider struct{}
+
+func (d *dataSetDeleteOriginOlderProvider) FilterData(ctx context.Context, repository dataStore.DataRepository, dataSet *dataTypesUpload.Upload, dataSetData data.Data) (data.Data, error) {
+	filterableDataSetData := dataSetData.Filter(func(datum data.Datum) bool {
+		return slices.Contains(filterableDataSetDataTypes, datum.GetType())
+	})
+
+	if selectors := d.GetDataSelectors(filterableDataSetData); selectors != nil {
```

Comment:
This block has quadratic complexity and might not work very well if we are processing large batches of data. Should we log a warning if the data exceeds a certain size?

Reply:
Updated code to remove complexity.